swift-sdk/AuthManager.swift

```diff
@@ -51,16 +56,43 @@ class AuthManager: IterableInternalAuthManagerProtocol {
     
     func retrieveAuthToken() {
         authToken = localStorage.authToken
+        
+        queueAuthTokenExpirationRefresh(authToken)
     }
     
     // MARK: - Private/Internal
     
+    private static let refreshWindow = 60
+    
+    private var expirationRefreshTimer: Timer?
+    
     private var authToken: String?
     
     private var localStorage: LocalStorageProtocol
     
     private let onAuthTokenRequestedCallback: (() -> String?)?
     
+    private func queueAuthTokenExpirationRefresh(_ authToken: String?) {
+        guard let authToken = authToken, let expirationDate = AuthManager.decodeExpirationDateFromAuthToken(authToken) else {
+            return
+        }
+        
+        let refreshDate = TimeInterval(expirationDate - AuthManager.refreshWindow)
```

Comment:
I am not sure what `refreshDate` is here. Is this a date or time interval?

Reply:
(same as above)